kyaml/fn/framework/example_filter_GVK_test.go

```diff
@@ -0,0 +1,34 @@
+package framework_test
+
+import (
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+)
+
+// This example implements a function that reads the desired replicas from the
+// functionConfig and updates the replicas field for all deployments.
+
+func Example_filterGVK() {
+	if err := command.AsMain(framework.ResourceListProcessorFunc(updateReplicas)); err != nil {
+		os.Exit(1)
+	}
+}
+
+// updateReplicas sets a field in resources selecting by GVK.
+func updateReplicas(rl *framework.ResourceList) error {
+	if rl.FunctionConfig == nil {
+		return framework.ErrMissingFnConfig{}
+	}
+	replicas, found, err := rl.FunctionConfig.GetNestedInt("replicas")
+	if !found || err != nil {
+		return err
+	}
+	for i, obj := range rl.Items {
+		if obj.GetApiVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
```

Comment:
Use the framework's matchers for this: `framework.GVKMatcher("apps/v1/Deployment").Match(items[i])`
(same for other similar places--we have lots of stock matchers written, and even ways to combine them easily).

Reply:
Thanks for bringing this up. `GVKMatcher().Match()` works here.
But thinking more about it, why do we make it return a `ResourceTemplateMatcher`? The signature is `func GVKMatcher(names ...string) ResourceTemplateMatcher` and it gives me the impression that it's not what I want, since I don't need templating at all.
We (Google) have built many KRM functions. But we haven't use this `ResourceTemplateMatcher` at all. I wonder what is the use cases for that? And do we really need it? `ResourceMatcher` seems to be good enough.